posydon/popsyn/binarypopulation.py

```diff
@@ -375,20 +407,8 @@ def save(self, save_path, **kwargs):
                 file_path = os.path.join(dir_name, file_name)
                 warnings.warn('The provided path is a directory - saving '
                               'to {0} instead.'.format(file_path), Warning)
-
-            self.comm.Barrier()
-
-            if self.rank == 0:
-
-                file_name = os.path.basename(absolute_filepath)
-                tmp_files = [os.path.join(
-                    self.kwargs["temp_directory"], f"evolution.combined.{i}")
-                             for i in range(self.size)]
-
-                self.combine_saved_files(absolute_filepath, tmp_files, mode=mode, **kwargs)
-
-            else:
-                return
+                
+            self.combine_save_files(absolute_filepath, temp_directory, file_name, **kwargs)
```

Comment:
@maxbriel I ran into an error here and I think this function call was supposed to be `combine_save**d**_files`, although there are too many arguments here for that method.

At some point we had code running in the notebooks with explicit calls to this save method when we were running individual binary populations at a time instead of straight to the synthetic population like:
```python
pop = BinaryPopulation(...)
pop.evolve(tqdm=True)
pop.save('filename.h5')
```
But after looking through the `_safe_evolve` it looks like saving is automatically done regardless of whether you call the save method, and instead this method is a combination step now?

Reply:
Actually I think I confused myself, it looks like in the non-parallel case it does actually save, but with MPI it will combine files saved in `_save_evolve`?

Reply:
Yeah that's definitely a wrong function reference and it should call `combined_saved_files`.
Although I think the function call should be  `self.combine_saved_files(absolute_filepath, tmp_files, mode=mode, **kwargs)`, since it will combine files from multiple-parallel runs.

with
       `tmp_files = [os.path.join(self.kwargs["temp_directory"], f"evolution.combined.{i}") for i in range(self.size)]

I haven't changed the structure of the `save`/`evolve` functions with respect to the how it save/stores data, but evolve is definitely saving the binaries while evolving them if the `ini` parameters are set so.

I've only removed the step in `SyntheticPopulation.evolve()` that called `BinaryPopulation.save`, because that it meant a sub-process had to wait for all processes to finish. Now it can move to the next metallicity and merging of files from reach sub-process happens after the runs have finished.

Reply:
@ka-rocha  I've altered this part of the code to read the temporary file directory and merge the files in that folder together.

`evolve` runs and saves the population if the right parameters are set (see `_safe_evolve`).
In the case of a single run, the single `evolution.combined` file is taken and written to a file with a given filename what you want it to be. 
In the case of an MPI/job array run, there will be multiple `evolution.combined_{i}` files, save should then take these and combine them into a file with your given file name :)

Reply:
Alright this looks good!